landa/water_body_management/utils.py

```diff
@@ -0,0 +1,195 @@
+import json
+
+import frappe
+from frappe import _
+from frappe.query_builder.custom import ConstantColumn
+from frappe.utils import cint
+
+
+def create_version_log(doc, event):
+	"""Called via hooks.py to create Version Log of document creation"""
+	if doc.doctype == "File" and doc.attached_to_doctype != "Water Body":
+		return
+
+	doc.flags.updater_reference = {
+		"doctype": doc.doctype,
+		"docname": doc.name,
+		"event": "Created",
+		"label": _("{0} {1} Created").format(doc.doctype, doc.name),
+	}
+
+	doc._doc_before_save = None
+
+
+def get_changed_data(from_datetime: str):
+	"""
+	Get created, modified and deleted Water Body, Fish Species, WBMLO, Attached File data.
+	"""
+	created_modified = get_version_log_query(from_datetime)
+	deleted = get_deleted_document_query(from_datetime)
+	union_query = created_modified.union(deleted)
+	return (frappe.qb.from_(union_query).select(union_query.star).orderby(union_query.creation)).run(
+		as_dict=True
+	)
+
+
+def get_formatted_changes(changes_list: list):
```

Comment:
Using  the `list` type here shouldn't work with python3.7. We can use from `typing import List` instead.